### PR TITLE
fix: Update API's to work for OpenShift 4.9

### PR DESCRIPTION
**Pull request by JPinkney**
### What does this PR do?
This PR builds off of https://github.com/devfile/devworkspace-operator/pull/540 to make everything work for both OpenShift 4.8 and 4.9

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v7-devworkspaces-operator-e2e, v7-devworkspace-happy-path` to trigger)
    - [ ] `v7-devworkspaces-operator-e2e`: DevWorkspace e2e test
    - [ ] `v7-devworkspace-happy-path`: DevWorkspace e2e test

### CODE REVIEW

File: webhook/workspace/mutating_cfg.go [hunk 1]
```diff
@@ -23,54 +23,56 @@ import (
...
-				Rule: v1beta1.Rule{
+				Operations: []admregv1.OperationType{admregv1.Create, admregv1.Update},
+				Rule: admregv1.Rule{
 					APIGroups:   []string{"controller.devfile.io"},
 					APIVersions: []string{"v1alpha1"},
 					Resources:   []string{"devworkspaceroutings", "components"},
 				},
 			},
 		},
+		AdmissionReviewVersions: []string{"v1beta1", "v1"},
```
Comment (sleshchenko) [inline webhook/workspace/mutating_cfg.go:68]: why it's needed? when should we remove it?

**Comment by sleshchenko** [comment]
happy path test fails due https://github.com/devfile/devworkspace-operator/issues/484

**Comment by sleshchenko** [comment]
waiting for my OpenShift 4.8, 4.9 clusters to test these changes.

File: webhook/workspace/mutating_cfg.go (hunk 1 above)
Comment (JPinkney) [reply webhook/workspace/mutating_cfg.go:68]: IIRC On OpenShift 4.9 it failed without it. I'll double check again though

Comment (sleshchenko) [reply webhook/workspace/mutating_cfg.go:68]: OK, seems I reproduced the issue you talk about on OpenShift 4.9
> Error from server (InternalError): error when creating "./samples/flattened_theia-next.yaml": Internal error occurred: failed calling webhook "mutate.devworkspace-controller.svc": received invalid webhook response: expected response.uid="07b5b66c-0d84-48bb-939f-c4f6bdcaf3a5", got ""

I assume we should be able to remove it after we upgrade controller-runtime in Angel's PR.

**Review by sleshchenko — APPROVED** [review]
Tested on minikube v1.22.0, OpenShift 4.8 and 4.9 and everything seems to work fine.

**Pull request merged**